frontend/passkeys.js

```diff
@@ -0,0 +1,35 @@
+document.addEventListener('DOMContentLoaded', function() {
+    'use strict';
+
+    const registerPasskeyButtons = document.querySelectorAll('#register-passkey');
+
+
+    async function registerPasskey() {
+        const response = await fetch('/passkey/registration-options') //fetch PublicKeyCredentialCreationOptions as JSON string
+        const publicKeyCredentialCreationOptionsJSON = await response.json() // convert to JSONObject
+        const credentialCreationOptions = PublicKeyCredential.parseCreationOptionsFromJSON(publicKeyCredentialCreationOptionsJSON); // convert to PublicKeyCredentialCreationOptions
+        const publicKeyCredential = await navigator.credentials.create({publicKey: credentialCreationOptions}); // create PublicKeyCredential
+        console.log('publicKeyCredential: ', publicKeyCredential);
+        const registrationResponseJSON = publicKeyCredential.toJSON(); // convert to JSONObject
+        await fetch('/passkey/register', {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'text/plain',
+                'Csrf-Token': `${registerPasskeyButtons[0].getAttribute('csrf-token')}`
```

Comment:
I wonder if it would be more elegant to pass the value of csrf-token into `registerPasskey` from the event listener.  We know that there will be a `registerPasskeyButtons[0]` at this point because otherwise `registerPasskey` wouldn't be called but it still seems a bit odd.